server/src/main/java/io/crate/planner/operators/WindowAgg.java

```diff
@@ -97,6 +97,28 @@ public static LogicalPlan create(LogicalPlan source, List<WindowFunction> window
                 lastWindowAgg = new WindowAgg(eval, windowDefinition, functions, eval.outputs());
             }
         }
+        /* Approach 2:
+
+         if first partition (that might be used in case of distributed execution), is not in the source plan outputs,
+         add that symbol to the outputs.
+
+         ie for the query
+
+         SELECT ROW_NUMBER() OVER (
+                PARTITION BY obj ['a']
+                ) AS RK
+         FROM (
+                SELECT obj
+                FROM test
+                ) t
+
+         we have outputs: List.of(obj)
+         and partitionSymbol obj ['a'].
+
+         turn outputs to
+         List.of(obj, obj ['a'])
+         */
```

Comment:
This is better then approach 1 that distrobuted column points to the exact column (and not the whole object) so we stream less (?) but we have redundant output.

Also, looks like we opted out of such output at least once, from https://github.com/crate/crate/pull/9529

>An alternative approach to fix this was in https://github.com/crate/crate/pull/9519

>Which changed the obj['x'] lookup on the sub-relation to change the
sub-relation itself.

>SELECT obj['x'] FROM (SELECT obj ...) AS t GROUP BY obj['x']

>would become

>SELECT obj['x'] FROM (SELECT obj, obj['x'] ...) AS t GROUP BY obj['x']
This approach looks simpler, but also seems to break semantics: It
introduces the notation that a child relation is modifying its outputs
due to the actions of a parent relation.